### DISCUSSION TITLE
Allow password modification in the UI

### DISCUSSION THREAD

**Discussion by Talanor** [jesec/flood#158] [Ideas]
Hey,
thanks for the nice work, I have a few ideas for improvements that may or may not be interesting but I'll share them, the first is: Being able to change the current user's password in flood's UI under "Authentication".

It will make it possible to pre-setup an account and allow the end-user to change their password (in case of a shared flood instance for example)

**Comment by jesec** [comment]
Flood server does support that:
https://github.com/jesec/flood/blob/c279760160519049839e359fdd20dd58acd8da3e/server/models/Users.ts#L134-L146

Currently you can do that with `PATCH /api/auth/users/{username}` API:
https://github.com/jesec/flood/blob/c279760160519049839e359fdd20dd58acd8da3e/shared/schema/api/auth.ts#L28-L30

Flood API is quite easy to use. You can call Flood's APIs simply with `curl`:

```sh
# Auth and get auth token from cookie
curl -c auth.token -X POST -H "Content-Type: application/json" -d '{"username":"test","password":"old"}' http://127.0.0.1:3000/api/auth/authenticate

# Change user (everything can be changed, not only password)
curl -b auth.token -X PATCH -H "Content-Type: application/json" -d '{"username":"newName","password":"newPassword"}' http://127.0.0.1:3000/api/auth/users/test
```

Frontend support is on the roadmap: https://github.com/jesec/flood/projects/5#card-51269106.

**Reply by Talanor** [reply]
Damn, I'll go read all of those. thanks again.
